internal/core/src/common/NullBitmapConcatenater.cpp

```diff
@@ -0,0 +1,71 @@
+#include "common/NullBitmapConcatenater.h"
+#include <cstdint>
+namespace milvus {
+void
+NullBitmapConcatenater::AddBitmap(uint8_t* bitmap, size_t size) {
+    bitmaps_.emplace_back(bitmap, size);
+}
+
+void
+NullBitmapConcatenater::ConcatenateBitmaps() {
+    size_t total_size = 0;
+    for (const auto& bitmap : bitmaps_) {
+        total_size += bitmap.second;
+    }
+    concatenated_bitmap_.first = new uint8_t[(total_size + 7) / 8];
+    concatenated_bitmap_.second = total_size;
+
+    size_t offset = 0;
+    for (const auto& bitmap : bitmaps_) {
+        if (bitmap.first == nullptr) {
+            // nullptr means nullable is false or no null value
+            // set offset to offset + bitmap.second bits are 0
```

Comment:
0 means the is not null right? null_bitmap_data taken from arrow, 0 means is null here.

Reply:
> 0 means the is not null right? null_bitmap_data taken from arrow, 0 means is null here.

oh yes, they use validity maps so 0 is invalid. I will fix it later.